pangeo_forge_recipes/transforms.py

```diff
@@ -638,3 +642,100 @@ def expand(
+    pyramid_kwargs: Optional[dict] = field(default_factory=dict)
+
+    def expand(self, pcoll: beam.PCollection) -> beam.PCollection:
+        return pcoll | beam.Map(
+            create_pyramid,
+            level=self.level,
+            epsg_code=self.epsg_code,
+            rename_spatial_dims=self.rename_spatial_dims,
+            pyramid_kwargs=self.pyramid_kwargs,
+        )
+
+
+@dataclass
+class StoreToPyramid(beam.PTransform, ZarrWriterMixin):
+
+    # Open design question. Should StoreToPyramid repeat all
+    # of the StoreToZarr kwargs and pass them through to the nested StoreToZarr
+    # or should it packaged them as store_to_zarr_kwargs and unpack?
+
+    """Store a PCollection of Xarray datasets as a Zarr Pyramid.
+    :param n_levels: Number of pyramid levels to generate
+    :param combine_dims: The dimensions to combine
+    :param store_name: Zarr store will be created with this name under ``target_root``.
+    :param other_chunks: Chunks for non-spatial dims.
+    Spatial dims are controlled by pixels_per_tile. Default is None
+      If a dimension is a not named, the chunks will be inferred from the data.
+    :param target_root: Root path the Zarr store will be created inside;
+        `store_name` will be appended to this prefix to create a full path.
+    :param epsg_code: EPSG code to set dataset CRS if CRS missing. If provided will overwrite.
+        Default is None.
+    :param rename_spatial_dims: Dict containing the new spatial dim names for Rioxarray.
+        Default is None.
+    :param pyramid_kwargs: Dict containing any kwargs that should be passed to ndpyramid.
+        Default is None.
+
+    """
+    n_levels: int
+    combine_dims: List[Dimension]
+    store_name: str
+    pixels_per_tile: Optional[int] = 128
+    other_chunks: Dict[str, int] = field(default_factory=dict)
+    target_root: Union[str, FSSpecTarget, RequiredAtRuntimeDefault] = field(
+        default_factory=RequiredAtRuntimeDefault
+    )
+    epsg_code: Optional[str] = None
+    rename_spatial_dims: Optional[dict] = None
+    pyramid_kwargs: Optional[dict] = field(default_factory=dict)
+
+    def expand(
+        self,
+        datasets: beam.PCollection[Tuple[Index, xr.Dataset]],
+    ) -> beam.PCollection[zarr.storage.FSStore]:
+
+        # Add multiscales metadata to the root of the target store
+        from ndpyramid.utils import get_version, multiscales_template
+
+        save_kwargs = {"levels": self.n_levels, "pixels_per_tile": self.pixels_per_tile}
+        attrs = {
+            "multiscales": multiscales_template(
+                datasets=[{"path": str(i)} for i in range(self.n_levels)],
+                type="reduce",
+                method="pyramid_reproject",
+                version=get_version(),
+                kwargs=save_kwargs,
+            )
+        }
+        chunks = {"x": self.pixels_per_tile, "y": self.pixels_per_tile}
+        if self.other_chunks is not None:
+            chunks |= self.other_chunks
+
```

Comment:
This is blocked by https://github.com/pangeo-forge/pangeo-forge-recipes/issues/520